index.html

```diff
@@ -1832,12 +1755,17 @@ <h3>Key types and formats</h3>
 }
         </pre>
 
-        <p>
-If a public key does not exist in the <a>DID document</a>, the key was revoked
-or is invalid. The <a>DID document</a> MUST NOT express revoked keys using a
-<a>verification relationship</a>. Each <a>DID method</a> specification is
-expected to detail how revocation is performed and tracked.
-        </p>
+
+<p>The <a>DID document</a> MUST NOT express revoked keys using a
+<a>verification relationship</a>.
+</p> 
+<p>
+If a verification method does not exist in the <a>DID document</a>, the method was revoked
+or is invalid. 
```

Comment:
I want to make sure we don't introduce any spec contradictions here that were clearly not intended.

First, this statement could be read to mean that any verification method that isn't referenced in the DID Document is invalid -- thus invalidating everyone else's verification methods and transitively your own (obviously absurd, but this is spec text, so we should be careful).

Second, a DID Document may have a `controller` property that refers to another DID Document in which there are verification methods that could be used to satisfy proofs that would be considered equivalent to those provided by the DID subject. We don't want readers to get confused where one section of the spec says this is ok, and a simple reading here says it's not. Again, that wasn't the intent here.

Lastly, I don't think we want to rule out use cases where other verification methods are expressed via VCs (e.g., for privacy purposes) but are ultimately rooted by a verification method in the DID Document. What we want to ensure is that the base or root is always the DID Document (or a DID Document referenced as a `controller`).

This is my attempt at saying that simply:

```suggestion
Proofs provided by the <a>DID subject</a> must be verifiable by a verification method
that is in the <a>DID Document</a>. This does not preclude the use of protocols that
involve chains of proofs provided that the base proof is verifiable by a verification
method that is in the <a>DID Document</a>. If the verification method for verifying
a base proof provided by the <a>DID subject</a> is not in the <a>DID Document</a>,
the verification method was revoked or is invalid. See <a href="#did-controller"></a> for
more information on verification methods for proofs that are not provided by the
<a>DID subject</a>.
```

Reply:
@OR13, if this suggestion is accepted, I'll approve, this is great. Thanks!

Reply:
I don't think I can accept this change, I didn't add anything regarding proofs, that statement regarding absence of a verification method was the result of working group consensus, and is logically simpler than your proposed text.

I don't think we should be describing "proofs" in general in did core, and the sentence as written does not prevent your proposed text.

I think you are opening a can of worms regarding proofs, which is not needed.

whether the vm is for a "base proof (undefined term) or proof chain (undefined term)" is not relevant. if a single verification method `id` is not present in the did document, the verification method (not the proof) is considered revoked.

Reply:
@OR13, the "proofs" language I used here mirrors the language used in the controller section. That's why I chose it -- to make it clear that we're not creating a contradiction between two different sections of the spec. That language also has consensus from the group.

Reply:
>  if a verification method id is not present in the did document, the verification method (not the proof) is considered revoked.

This statement is problematic because it presumes that the verification method ID was, at some point, *in* the DID Document and because it makes it sound like a verification method that is in an entirely different DID Document is invalid or revoked simply because it is not present in the DID Document you're looking in. This is why referring to proofs is better -- the whole reason you're looking at a verification method is because you're dealing with a thing (a proof of some sort) that you're trying to verify. If that proof has led you to the current DID Document, and the verification method isn't there, then it cannot be used to verify that proof. That's the concept we want to get across here.

Reply:
Resolving this conversation and referencing a new suggestion here: https://github.com/w3c/did-core/pull/490#discussion_r543618406